### PR TITLE
Refactor masking

### PR DESCRIPTION
**Pull request by ajaitasaini**
### Description

Masking numerical user inputs in MCR has gotten complex, and requires a refactor to decrease code bloat + excessive reliance on masking over input validations.

More details [here](https://docs.google.com/document/d/1d0L2UyKcCU0lUJBFvGHJUJ-UIdredIasIHLO17a_wcM/edit)

### Related ticket(s)

MDCT-2597
MDCT-2665

---
### How to test

1. Login to MCR
2. Open either MCPAR or MLR, and navigate to a numerical input field 
3. Test inputs for validity (example valid/invalid inputs listed [here](https://docs.google.com/document/d/1d0L2UyKcCU0lUJBFvGHJUJ-UIdredIasIHLO17a_wcM/edit#heading=h.qgirlr2c8bqw))
4. If value is valid, confirm that it is getting masked correctly, and the cleaned value is getting sent to the database

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/utils/other/checkInputValidity.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  if (
+    !checkLeftSideOfRatioAgainstRegexes(values[0]) ||
+    !checkRightSideOfRatioAgainstRegexes(values[1])
+  )
+    return { isValid: false, leftSide: values[0], rightSide: values[1] };
+
+  return { isValid: true, leftSide: values[0], rightSide: values[1] };
+};
+
+export const checkLeftSideOfRatioAgainstRegexes = (value: string): boolean => {
```
Comment (oweingrod-coforma) [inline services/ui-src/src/utils/other/checkInputValidity.ts:62]: Would it be possible to combine these functions if the regex is the same?

File: services/ui-src/src/utils/other/mask.tsx [hunk 2]
```diff
@@ -1,104 +1,88 @@
...
+export const applyMask = (
   value: string,
-  mask: keyof typeof customMaskMap
-): string {
-  const selectedCustomMask = customMaskMap[mask];
-  if (isValidNumericalString(value)) {
-    return selectedCustomMask(value);
-  } else return value;
-}
+  maskName?: keyof typeof maskMap | null
```
Comment (oweingrod-coforma) [inline services/ui-src/src/utils/other/mask.tsx:75]: What happens if `maskName` is undefined here?

**Review by oweingrod-coforma — COMMENTED** [review]
Generally looks good, just two quick questions/ideas!

File: services/ui-src/src/utils/other/clean.test.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { cleanStandardNumericalInput, cleanRatioInput } from "./clean";
+
+const cleanStandardNumericalInputTestCases = [
+  { test: "0....0123", expected: "0....0123" },
+  { test: "000000123", expected: "000000123" },
```
Comment (keeysnc) [inline services/ui-src/src/utils/other/clean.test.tsx:5]: For this one should the expected value be 123?

**Review by keeysnc — APPROVED** [review]
Looks good to me, the cleaned values are being send to the db.

File: services/ui-src/src/utils/other/checkInputValidity.ts (hunk 1 above)
Comment (ajaitasaini) [reply services/ui-src/src/utils/other/checkInputValidity.ts]: Good call out, thank you! combined the functions

File: services/ui-src/src/utils/other/mask.tsx (hunk 2 above)
Comment (ajaitasaini) [reply services/ui-src/src/utils/other/mask.tsx:75]: it defaults to the thousand separated mask!

**Review by karla-vm — APPROVED** [review]
Looks great!

**Review by karla-vm — APPROVED** [review]

**Review by oweingrod-coforma — APPROVED** [review]

**Pull request merged**